lib/listener.js

```diff
@@ -351,6 +351,40 @@ internals.Listener.prototype._subscribe = function (path, socket, next) {
 };
 
 
+internals.Listener.eachSocket = function (callback, options) {
+
+    options = options || {};
+
+    var connections = this.connections;
+    for (var i = 0, il = connections.length; i < il; ++i) {
+        var connection = connections[i];
+        if (connection.plugins.nes) {
+            connection.plugins.nes._listener._eachSocket(callback, options);
+        }
+    }
+};
+
+
+internals.Listener.prototype._eachSocket = function (callback, options) {
+
+    var subscription = options.subscription;
+    if (!subscription) {
+        return this._sockets.forEach(callback);
+    }
+
+    var sub = this._router.route('sub', subscription);
+    if (sub.isBoom) {
+        return;
+    }
+
+    var route = sub.route;
+    route.subscribers.forEach(sub.paramsArray.length ? subscription : null, function (socket) {     // Filter on path if has parameters
```

Comment:
This is for filtering to any parameters defined in the subscription path. Similar code exists in the publish flow, but the variable is called `path` there.